nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/AbstractAzureDataLakeStorageProcessor.java

```diff
@@ -146,8 +154,22 @@ public static DataLakeServiceClient getStorageClient(PropertyContext context, Fl
         return storageClient;
     }
 
-    @Override
-    public Set<Relationship> getRelationships() {
-        return RELATIONSHIPS;
+    private static class DirectoryValidator implements Validator {
+        @Override
+        public ValidationResult validate(String subject, String input, ValidationContext context) {
+            ValidationResult.Builder builder = new ValidationResult.Builder()
+                    .subject(DIRECTORY.getDisplayName())
+                    .input(input);
+
+            if (context.isExpressionLanguagePresent(input)) {
+                builder.valid(true).explanation("Expression Language Present");
```

Comment:
A runtime check (in onTrigger) for the actual value (after evaluating the expression language) might be necessary.

Reply:
Added runtime check.
Also moved the runtime check methods to the abstract parent class.